examples/bsearch.masm

```diff
@@ -0,0 +1,208 @@
+#! This example code performs binary searching for addr given value in an ascending ordered array.
+#!
+#! Input must be provided through the operands stack.
+#! For input and output see documentation for `proc.binary_search_stack`.
+#!
+#! If input values are omitted or `count = 0`, tests will run.
+
+const.STARTING_MEMORY_ADDRESS=0
+
+#! Binary searches for value `val` in the memory addresses range [start..end), sorted in ascending order.
+#!
+#! Input: [start, end, val, ...]
+#! Output: [found (true/false), addr] where `found` is searching result (`true`/`false`),
+#!         `addr` is an address of value (if it was found), or an address where value should be inserted.
+proc.binary_search
+    # Check if `start < end` then start search, otherwise prepare "not found" result and exit
+    u32assert2                              # [start, end, val, ...]
+    dup                                     # [start, start, end, val, ...]
+    dup.2                                   # [end, start, start, end, val, ...]
+    u32checked_lt                           # [start < end, start, end, val, ...]
+    if.true                                 # [start, end, val, ...]
+        push.1                              # [1, start, end, val, ...]
+    else                                    # [start, end, val, ...]
+        # Push "not found" flag and flag to skip the loop
```

Comment:
This would be an error, right? 

You could use `assert.err=123` and break.

Reply:
This is not an error. Binary search is used not only for searching, but also for adding new values into a sorted array. In a case when value wasn't found, the algorithm must return an index where this value should be inserted.